app/src/main/java/com/zarinraim/dogsearch/presentation/main/all_breeds/BreedsListScreen.kt

```diff
@@ -48,11 +54,11 @@ fun BreedsListScreen(
 @ExperimentalMaterialApi
 @Composable
 fun BreedsList(
-    breeds: Map<String, List<String>>,
-    navController: NavController,
-    listState: LazyListState
+    breeds: AllBreeds,
+    listState: LazyListState,
+    onClickOpenImage: (String, String) -> Unit
 ) {
-    val mainBreedsList = breeds.keys.toList()
+    val mainBreedsList = breeds.map.keys.toList()
```

Comment:
This conversion can be done inside the view model. Generally speaking, ViewModel should prepare data to show.